web/app/components/billing/billing-page/index.tsx

```diff
@@ -1,28 +1,24 @@
 'use client'
 import type { FC } from 'react'
-import React, { useEffect } from 'react'
+import React from 'react'
 import { useTranslation } from 'react-i18next'
+import useSWR from 'swr'
 import PlanComp from '../plan'
 import { ReceiptList } from '../../base/icons/src/vender/line/financeAndECommerce'
 import { LinkExternal01 } from '../../base/icons/src/vender/line/general'
 import { fetchBillingUrl } from '@/service/billing'
 import { useAppContext } from '@/context/app-context'
 import { useProviderContext } from '@/context/provider-context'
 
+
 const Billing: FC = () => {
   const { t } = useTranslation()
   const { isCurrentWorkspaceManager } = useAppContext()
-  const [billingUrl, setBillingUrl] = React.useState('')
   const { enableBilling } = useProviderContext()
-
-  useEffect(() => {
-    if (!enableBilling || !isCurrentWorkspaceManager)
-      return
-    (async () => {
-      const { url } = await fetchBillingUrl()
-      setBillingUrl(url)
-    })()
-  }, [isCurrentWorkspaceManager])
+  const { data: billingUrl } = useSWR(
+    !enableBilling || !isCurrentWorkspaceManager ? null : ['/billing/invoices'],
```

Comment:
Please use this function https://github.com/langgenius/dify/blob/b521aafd26ee95d91bd48660c515e572697bb1ec/web/service/billing.ts#L13

Or use the const here instead of `'/billing/invoices'` in this file.

Reply:
![image](https://github.com/langgenius/dify/assets/42225395/2fdb6ad1-11c9-4aca-9bf2-5b294a044357)
I did use the fetch billing function, I'm using the api path as the query key in the first argument in useSWR. Is this okay?

Reply:
Maybe you can make it as a constant.